internal/kubernetes/watch.go

```diff
@@ -28,29 +62,49 @@ type watcher struct {
 	secrets map[types.NamespacedName][]*corev1.Pod
 	// secretKeysByPod enables reverse lookup Secrets keys per Pod.
 	secretKeysByPod map[types.NamespacedName][]types.NamespacedName
+
+	cmLister     listersv1.ConfigMapLister
+	secretLister listersv1.SecretLister
+}
+
+var _ PodResourceManager = (*watcher)(nil)
+
+func NewPodResourceWatcher(informerFactory informers.SharedInformerFactory) PodResourceManager {
+	return newPodResourceWatcher(informerFactory)
 }
 
-func newWatcher() *watcher {
+func newPodResourceWatcher(informerFactory informers.SharedInformerFactory) *watcher {
 	return &watcher{
 		configs:         make(map[types.NamespacedName][]*corev1.Pod),
 		cmKeysByPod:     make(map[types.NamespacedName][]types.NamespacedName),
 		secrets:         make(map[types.NamespacedName][]*corev1.Pod),
 		secretKeysByPod: make(map[types.NamespacedName][]types.NamespacedName),
+		cmLister:        informerFactory.Core().V1().ConfigMaps().Lister(),
+		secretLister:    informerFactory.Core().V1().Secrets().Lister(),
 	}
 }
 
-func (w *watcher) handlerFuncs(ctx context.Context, p updater) cache.ResourceEventHandlerFuncs {
+func (w *watcher) ConfigMapLister() listersv1.ConfigMapLister {
+	return w.cmLister
```

Comment:
Just make these fields public, no need for 1-line accesor functions

Reply:
That doesn't work when ctors return an interface impl. If I change the ctor, then I cannot use the interface to pass this to provider. I wanted to separate what is pure Kubernetes stuff like this from what is provider.

Reply:
Miek's gave a nod to my last comment https://github.com/virtual-kubelet/systemk/pull/65#issuecomment-763473080